### PR TITLE
feature/added-functionality-to-csvfiles

### PR DESCRIPTION
**Pull request by Pallavi0717**

### CODE REVIEW

File: phase1/modlix db/main.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import os
+import pandas as pd
+from database import get_engine, create_table
+from file_processing import list_csv_files, infer_table_schema, load_csv_to_db
+
+def main():
+    user = 'root'
+    password = quote('A1b2c3@123')
+    host = 'localhost'
+    directory_path = 'venv\\modlixdata1'
```
Comment (lawbringr1) [inline phase1/modlix db/main.py:11]: what will happen we are using UNIX path? using /

File: phase1/modlix db/database.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from sqlalchemy import create_engine, MetaData, Table 
+
+def get_engine(user, password, host, database):
+    return create_engine(f'mysql+mysqlconnector://{user}:{password}@{host}/{database}')
+
+def create_table(engine, table_name, columns):
+    metadata = MetaData()
+    table = Table(table_name, metadata, *columns)
+    metadata.create_all(engine)
+    print(f"Table '{table_name}' created or already exists.")
```
Comment (lawbringr1) [inline phase1/modlix db/database.py:10]: no print statement only log

File: phase1/modlix db/main.py (hunk 1 above)
Comment (lawbringr1) [inline phase1/modlix db/main.py:11]: no secret sharing on github

File: phase1/modlix db/main.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        file_path = os.path.join(directory_path, filename)
+        table_name = filename[:-4]  # Removing the '.csv' extension
+
+        sample_chunk = pd.read_csv(file_path, nrows=1000)
+        columns = infer_table_schema(sample_chunk)
+        
+        create_table(engine, table_name, columns)
+        load_csv_to_db(engine, file_path, table_name)
+    
+    print(f"Finished loading {len(csv_files)} CSV files into the MySQL database '{database}'.")
```
Comment (lawbringr1) [inline phase1/modlix db/main.py:26]: not print only log

File: phase1/modlix db/file_processing.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import os
+import pandas as pd
+from sqlalchemy import Column, Integer, String, Float, BigInteger, DateTime, TEXT
+
+def list_csv_files(directory_path):
```
Comment (lawbringr1) [inline phase1/modlix db/file_processing.py:5]: list means creating list of listing list?

File: phase1/modlix db/file_processing.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import os
+from pathlib import Path
+import pandas as pd 
+from sqlalchemy import Column, Integer, String, Float, BigInteger, DateTime, TEXT 
+from sqlalchemy.exc import SQLAlchemyError
+import logging
+
+def list_csv_files(directory_path):
+    supported_extensions = ('.csv', '.data.csv')
```
Comment (lawbringr1) [inline phase1/modlix db/file_processing.py:9]: .data.csv is not an extension

File: phase1/modlix db/main.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import pandas as pd
+from database import get_engine, create_table
+from file_processing import list_csv_files, infer_table_schema, load_csv_to_db
+logging.basicConfig(level=logging.INFO)
+
+def main():
+    user = '****'
+    password = quote('********')
+    host = '*******'
+    directory_path = Path('venv\modlixdata1')
```
Comment (lawbringr1) [inline phase1/modlix db/main.py:14]: get current working path

File: phase1/modlix db/main.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    database = directory_path.name
```
Comment (lawbringr1) [inline phase1/modlix db/main.py:15]: what if directory is not there?

Comment (lawbringr1) [reply phase1/modlix db/main.py:15]: throw an error if the data folder is not having proper structure

**Comment by Pallavi0717** [comment]
features/added-changes-to-csvfiles

**Comment by Pallavi0717** [comment]
feature/added-changes-to-csvfiles

**Pull request closed without merging**